### PR TITLE
Hack around missing 'ontransitionend' event on zoom anim

### LINKED ISSUE #2693: Map freeze problem, but unlocks after marker click
Hello

I'm using leaflet 0.7.2 in a windows RT application, developed with the use of the WinJS library. I'm also using angularjs as the basis for the application. The map freezes 1/20 times when i zoom in or out. When i press a marker on the map, the map unfreezes and becomes responsive once more. When i use the leaflet-src and debug the problem i notice, that when the problem occurs the "_catchTransitionEnd"  isn't called. When i press a marker the "_catchTransitionEnd" is called and the map unfreezes. Does anyone know what is causing this problem, and better still how to fix it? a quick workaround would be great, because of the deadline i'm on.

Any help would be greatly appreciated.

Thanks in advance

Ian

### PR DESCRIPTION
**Pull request by IvanSanchez**
Should fix #2693

### CODE REVIEW

File: src/map/anim/Map.ZoomAnimation.js [hunk 1]
```diff
@@ -104,6 +104,14 @@ L.Map.include(!zoomAnimated ? {} : {
...
 			noUpdate: noUpdate
 		});
+
+
+		// This shouldn't be needed if browsers fired ontransitionend consistently
+		setTimeout(L.bind(function(){
+			if (this._animatingZoom) {
+				this._onZoomTransitionEnd();
+			}
+		}, this), 630);
```
Comment (yohanboniface) [inline src/map/anim/Map.ZoomAnimation.js]: Where is 630 from? :)

File: src/map/anim/Map.ZoomAnimation.js [hunk 2]
```diff
@@ -104,6 +104,14 @@ L.Map.include(!zoomAnimated ? {} : {
 			zoom: zoom,
 			noUpdate: noUpdate
 		});
+
+
+		// This shouldn't be needed if browsers fired ontransitionend consistently
+		setTimeout(L.bind(function(){
```
Comment (yohanboniface) [inline src/map/anim/Map.ZoomAnimation.js]: eslint does not complain about the missing space around `()`?

Comment (jieter) [inline src/map/anim/Map.ZoomAnimation.js]: It does in travis:
2015-10-06-111923_699x60_scrot

**Comment by IvanSanchez** [comment]
@yohanboniface The decision to use `630` msec comes from checking times with `performance.now()` between the `touchend` and `transitionend` events - it was around 620-625.

**Comment by yohanboniface** [comment]
Shouldn't we take a margin? I guess this time will be different for different CPUs or not?

**Comment by IvanSanchez** [comment]
@yohanboniface I'm doing stuff with an iPad 2 (2011!) so it should be slow enough. Do feel free to experiment with `debug/hacks/ontransitionend.html`, check out the timings in the debug console, and see if you can come up with a different number.

**Comment by mourner** [comment]
This looks very similar to the hack I did for 0.7.5, but firing `setTimeout` at the same time as launching the transition to clean up after transitionend. This seems less fishy than having a magic 630ms to fire after touchend. Maybe we should just port over the fixes from stable then?

**Comment by IvanSanchez** [comment]
@mourner Sure, sounds like a good idea. Is there a PR or commit with the stable fix I can have a look at?

**Comment by mourner** [comment]
@IvanSanchez https://github.com/Leaflet/Leaflet/commit/91477588faa44e2b266d86a5a44cb98490244d39, https://github.com/Leaflet/Leaflet/commit/19c295ea7cc27f6aaaabed7c14d4affc8113279f, https://github.com/Leaflet/Leaflet/commit/3e200bed22bdb1f59e54f37c93b4fce034fd7ee4

**Pull request closed without merging**